File: genetic-packer/src/main/java/genetic/packer/Runner.java
package genetic.packer;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

/**
 * @author piotr.larysz
 */
@Component
public class Runner implements CommandLineRunner {

    @Autowired
    private Runnable fitnessLoggingAdapter;

    @Override
    public void run(String... strings) throws Exception {

        this.fitnessLoggingAdapter.run();
    }
}

File: genetic-packer/src/main/java/genetic/packer/Runner.kt
package genetic.packer

import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class Runner(private val fitnessLoggingAdapter: Runnable) : CommandLineRunner {

  override fun run(vararg strings: String) {
    fitnessLoggingAdapter.run()
  }
}